src/App.tsx

```diff
@@ -1,16 +1,87 @@
 import React from 'react';
 import './App.scss';
 
-class App extends React.Component {
-  addOne = () => {};
+const colors = [
+  { id: 1, color: 'yellow', hex: '#FFFF00' },
+  { id: 2, color: 'orange', hex: '#FFA500' },
+  { id: 3, color: 'red', hex: '#FF0000' },
+];
 
-  add100 = () => {};
+interface State {
+  counter: number,
+  index: number,
+}
+
+class App extends React.Component<{}, State> {
+  state = {
+    counter: 0,
+    index: 0,
+  };
+
+  componentDidUpdate() {
+    setTimeout(() => {
+      if (this.state.counter > 99) {
+        this.setState({ index: 1 });
+      }
+
+      if (this.state.counter > 199) {
+        this.setState({ index: 2 });
+      }
+    }, 100);
+  }
+
+  addOne = () => {
+    this.setState((state) => ({
+      counter: state.counter + 1,
+    }));
+  };
 
-  increase = () => {};
+  add100 = () => {
+    this.setState((state) => ({
+      counter: state.counter + 100,
+    }));
+  };
+
+  increase = () => {
+    this.addOne();
+    if (this.state.counter % 5 === 0 && this.state.counter !== 0) {
```

Comment:
при першому кліку в нас зразу має бути 101 тому друга умова не потрібна)
```suggestion
    if (this.state.counter % 5 === 0) {
```